includes/indices/class-algolia-posts-index.php

```diff
@@ -430,4 +430,17 @@ protected function extract_item(Algolia_Task $task) {
 			
 		return get_post( $data['post_id'] );
 	}
+
+	/**
+	 * @param Algolia_Task $task
+	 */
+	public function delete_item( Algolia_Task $task ) {
+		if ( ! isset( $data['post_id'] ) || ! is_int( $data['post_id'] ) ) {
```

Comment:
This will trigger PHP notice undefined variable `$data`. We need to get data from task first using `$data = $task->get_data();`. Same applies to other index `delete_item()` methods.

Also it would be good practice to add abstract method delete_item to base abstract index class.